### DISCUSSION TITLE
Animating big amount of elements with delay without performance hits

### DISCUSSION THREAD

**Discussion by xzilja** [software-mansion/react-native-reanimated#4829] [Q&A]
I'm trying to re-create animation similar to one below, in essence multiple elements animate one after another with a delay. Some elements start animating before previous animation is finished, so I can't chain animation end callbacks.

The only way I was able to achieve this via was by using 10+ share animation values, each driving animated styles for individual elements, then I start animating them using `withDelay` and different delay values. But this causes a lot of performance issues on actual device.

So I am wondering, what would be the best / correct way to achieve something like below? Thank you

https://github.com/software-mansion/react-native-reanimated/assets/3154053/a1ae7be7-520c-4964-811d-8e3a9e2cd72f

**Comment by WadhahEssam** [comment]
this might sound like a stupid idea, but maybe you can create a timer and interpolate the animation to be based on the timer ms.

**Comment by xzilja** [comment]
I managed to solve this, lag was coming from using expo-image, I guess after initial install it reads normally from disc, but afterwards it's own cache, which seems to be lagging behind. Using normal rn image works. But I did end up with like 10 different shared values and `withDelay`s in my code.

I tried to interpolate on one value, but you loose individual "springiness" on each element that way. 

Might keep this open to get more ideas / suggestions.
